### PR TITLE
FRONT-1808: Goodbye trackers. Welcome crawler.

### PR DESCRIPTION
**Pull request by mondoreale**

### CODE REVIEW

**Comment by mondoreale** [comment]
Let's wait for GitHub to get their ducks in a row before going into a review or something.

File: config-overrides.js [hunk 1]
```diff
@@ -1,39 +1,49 @@
...
+  config.module.rules = updatedRules
 
   if (process.env.REACT_APP_PUBLIC_URL) {
     config.output.publicPath = `${process.env.REACT_APP_PUBLIC_URL}/`
   }
 
+  // Drop `ModuleScopePlugin`.
+  config.resolve.plugins = []
+
+  config.resolve.fallback = {
```
Comment (mondoreale) [inline config-overrides.js:41]: These fallbacks and the `process/browser` are required by the new `@streamr/utils`.

File: src/components/Debug/index.tsx
Comment (mondoreale) [inline src/components/Debug/index.tsx:1]: Doesn't seem needed.

File: src/components/Graphs/TimeSeries.tsx [hunk 2]
```diff
@@ -188,6 +182,24 @@ const UnstyledTimeSeriesGraph = ({
   )
 }
 
-const TimeSeries = styled(UnstyledTimeSeriesGraph)``
+export const TimeSeries = styled(UnstyledTimeSeriesGraph)``
```
Comment (mondoreale) [inline src/components/Graphs/TimeSeries.tsx:185]: I progressively switch to named exports and I remove things like
```ts
const Component = () => <></>

const Subcomponent = () => <></>

// This!
Object.assign(Component, { Subcomponent })
```

File: src/components/NodeList/NodeListItem.tsx [hunk 3]
```diff
@@ -111,56 +114,102 @@ const Address = styled(PlaceName)`
...
     leave: { opacity: 0 },
   })
 
+  const ref = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    let mounted = true
+
+    if (isActive) {
+      setTimeout(function scrollReffedElementIntoView() {
```
Comment (mondoreale) [inline src/components/NodeList/NodeListItem.tsx:150]: Each node list element takes care of its own "auto-scroll into view" behaviour. I dropped the `data-node-id` custom attr. Too much magic.

File: src/components/NodeList/NodeListItem.tsx [hunk 4]
```diff
@@ -111,56 +114,102 @@ const Address = styled(PlaceName)`
...
+           * If the page address includes the node id already and the user
+           * panned away then the above won't be sufficient to get node's location
+           * back into viewport. To address it, we have to invalidate the node id
+           * param key.
+           */
+          invalidateNodeIdParamKey()
+        }}
+        onMouseEnter={() => {
+          if (highlightPointOnHover) {
+            setNodeFeatureState(mapRef, nodeLocationId, { hover: true })
```
Comment (mondoreale) [inline src/components/NodeList/NodeListItem.tsx:194]: Hovering items on the list is reflected on the map now.

File: src/components/Place.tsx
Comment (mondoreale) [inline src/components/Place.tsx:1]: Plase takes node's coordinated and displays name of the place (using mapbox API, still).

**Review by mondoreale — COMMENTED** [review]
A round of comments.

**Comment by tumppi** [comment]
What a monster 👿  I had a quick look at the code and it surely looks good.

I tested it with Chrome but somehow it was hogging cpu like crazy. Even panning the map was somehow laggy compared to the old one? Have you experienced anything like this?

BTW, is the scroll zooming disabled on map intentionally? I used it all the time while surfing the network map :)

**Comment by mondoreale** [comment]
@tumppi, good feedback, cheers!

I'll fix the map interaction. I'm guessing it's a missing debounce.

The CPU usage may be related to… not being able to scope resends to a single sender (node address). We filter them within the browser. It's a problem that I'll raise tomorrow in the frontend call, too.

**Comment by mondoreale** [comment]
@tumppi the app is using the latest `react-map-gl` and `mapbox-gl`. It's a sizeable update but definitely worth it. No debouncing needed. Actually, I don't even manually touch the viewport props (lat, lng, zoom). All's done through the api exposed by the library.

Please have another look. It definitely is smoother!

**Review by tumppi — APPROVED** [review]
Now it feels much smoother :+1

**Pull request merged**